### ISSUE TITLE
Comas in DamageMultiplier, EffectBonusMultiplier & RealPointsPerLevel truncate the data when trying to copy the spell

### ISSUE THREAD

**Issue by Makpptfox** [stoneharry/WoW-Spell-Editor#66]
When you set a coma in the values of RealPointsPerLevel and/or EffectBonusMultiplier and/or DamageMultiplier the editor is unable to copy the spell if you want to use it as a base for a new spell, you get a truncated data error.

To reproduce this issue, simply create a new spell using the SpellID **205** as a base for it and you will get the error for RealPointsPerLevel. (You can test and change the value of DamageMultiplier and EffectBonusMultiplier for this spell and try to copy and you will be able to reproduce the two others errors)

I am using the lastest version of the Editor (1.8.2) in 64bits and have the French locale.

**Comment by stoneharry** [comment]
@Makpptfox  I think this is a locale issue. In France you use a comma as the decimal separator right? Use a full stop instead. For example `23.12` instead of `23,12`.

**Comment by Makpptfox** [comment]
When I do I can't save the spell I try to set full stop instead of a comma. We do use comma but we don't when using any software in general, since it's a very specific way of doing things, it is the same for our DBC afaik.

I tried anyway but get the error "Error Input string was not in a correct format" when I try to save.

I looked upon the enUS Spell.dbc and it is in fact also a comma, even when I'm opening it raw it is a comma, so it isn't some kind of automatic translation thingy from the editor either.

Here's a live test&proof : https://gfycat.com/UncomfortableTemptingLeafhopper

**Comment by stoneharry** [comment]
@Makpptfox I am still unfortunately not able to reproduce this. 😞 When saving with a full stop, it saves the value correctly, sometimes adjusting it because not all values can be represented with a float. When saving with a comma it actually converts it into a integer removing the comma.

https://www.dropbox.com/s/5oz89ld4zbvflps/2019-02-17_16-14-05.mp4?dl=0

Can you try deleting the `spell` table and try importing a fresh set of DBC's? It will automatically recreate the `spell` table. If you are using SQLite you will need to delete the `SpellEditor.db` file. I am wondering if one of your columns in the database has a different data type from an older version.

Let me know if you have any other ideas. 🤷‍♂️

**Comment by Makpptfox** [comment]
I don't have access to my DB right now, but I'll try to do so on a full clean database and dbc later. 
And if the problem persist, i'll try to switch my OS in english to see if it is the issue.

**Comment by barncastle** [comment]
Can confirm this is a OS localisation issue. To replicate I set the app's current culture to the french in the `MainWindow` constructor and copied spell id 205.
```c#
System.Threading.Thread.CurrentThread.CurrentCulture = new System.Globalization.CultureInfo("fr-FR");
```

If the database is always going to be using a decimal separator you could enforce this via code i.e.
```c#
var customCulture = (System.Globalization.CultureInfo)System.Threading.Thread.CurrentThread.CurrentCulture.Clone();
customCulture.NumberFormat.NumberDecimalSeparator = ".";
System.Threading.Thread.CurrentThread.CurrentCulture = customCulture;
```

**Comment by stoneharry** [comment]
Thanks barncastle! I did not know about this.

**Comment by stoneharry** [comment]
Included barncastle's fix in d166134. This will be included in release v1.8.3.

**Comment by stoneharry** [comment]
Released build v1.8.3 which includes this.
